torch_xla/csrc/tensor.cpp

```diff
@@ -506,7 +506,8 @@ void XLATensor::UpdateFromTensor(at::Tensor tensor, bool sync) {
     data()->handle = nullptr;
     AssignIrValue(torch::lazy::Value());
     if (data()->view != nullptr) {
-      torch::lazy::Value ir_value = GetIrValueForTensor(coyped_tensor, device);
+      torch::lazy::Value ir_value =
+          GetIrValueForTensor(coyped_tensor, GetDevice());
```

Comment:
Is the issue here that `data()->view` is being updated with placeholder IR? And even if we updated the `ir_value` with correct value(assuming virtual device take care of it), `data()->view` will be outdated?

@alanwaketan I thought `data()->view` is not being used if we enabled the functionization pass?

Reply:
Yeah that's correct - `data()->view` has placeholder IR if we use virtual device.

Reply:
I felt like this is not the right solution to this issue, maybe a better way is to update the view when we update the IR after virtual device.

let me look into virtual device code a bit.

Reply:
Unfortunately, we haven't cleaned up all the view op code in tensor_methods.cpp yet. @jonb377 If the issue is caused by narrow, let me clean up https://github.com/pytorch/xla/blob/master/torch_xla/csrc/tensor_methods.cpp#L1822 for you. Let me scope the clean up work for tensor_methods.cpp.

Reply:
I see, thanks Jiewen. I'm okay to cancel this PR in favor of functionalization.

The reason I caught this is it breaks resharding in distributed checkpointing, but as long as the device mesh stays the same distributed checkpointing works.

Reply:
We shouldn't need to deal with views after functionalization. Can you confirm the issue you saw is caused by narrow? Or other view ops? I need to prioritize it.

Reply:
This was caused by `narrow`, which is the only operation required for resharding. I didn't try any other view ops, but this would happen any time we `copy_` into a view.

Reply:
Got it. I can do the narrowing for you tomorrow. For other view ops, stay tuned on the plan.